oktaawscli/aws_auth.py

```diff
@@ -76,12 +76,18 @@ def choose_aws_role(self, assertion):
     def get_sts_token(self, role_arn, principal_arn, assertion, duration):
         """ Gets a token from AWS STS """
 
-        # Connect to the GovCloud STS endpoint if a GovCloud ARN is found.
-        arn_region = principal_arn.split(':')[1]
-        if arn_region == 'aws-us-gov':
-            sts = boto3.client('sts', region_name='us-gov-west-1')
-        else:
-            sts = boto3.client('sts', region_name=self.region)
+        try:
+            profile = os.environ.pop('AWS_PROFILE', None)
```

Comment:
nitpick: lets keep the profile logic together and move this and the `try` to right above line 87 so that we end up having the following

```python
        try:
            profile = os.environ.pop('AWS_PROFILE', None)
            if profile is not None:
                os.environ['AWS_PROFILE'] = profile
        except ProfileNotFound:
            self.logger.exception("Unable to handle AWS_PROFILE=%s" % os.environ['AWS_PROFILE'])
```

Reply:
ic... I missed the pop. can you add a succinct comment about that above the pop?

Reply:
can the ProfileNotFound exception also happen on the `boto3.client('sts', region_name=self.region)` call?

Reply:
Sure, but that's in side the try-except